lib/3scale/backend/storage_helpers.rb

```diff
@@ -291,7 +291,7 @@ def cfg_reconnect_handler(options)
             options[:reconnect_attempts] = if ThreeScale::Backend.listener?
                                              1
                                            else
-                                             ThreeScale::Backend.configuration.redis.async ? Float::INFINITY : 1
+                                             ThreeScale::Backend.configuration.redis.async ? Float::INFINITY : Hash.new(5)
```

Comment:
Is this documented? I understand we are going to remove sync anyway so if you tested it working, I guess it's alright. Thanks.

Reply:
It's documented [here](https://github.com/redis-rb/redis-client?tab=readme-ov-file#reconnection). It expects an array, but I wrote a hack to behave like an infinite array of `5`. Ugly but effective to match the behavior between sync and async modes.